base/pkg/entry.jl

```diff
@@ -212,6 +215,8 @@ function clone(url_or_pkg::AbstractString)
         pkg = url_or_pkg
         url = readchomp(urlpath)
         # TODO: Cache.prefetch(pkg,url)
+    elseif !(':' in url_or_pkg) && !isfile(urlpath)
+        throw(PkgError("package $url_or_pkg doesn't appear to exist"))
```

Comment:
I agree the LibGit2 module would probably be a better place to fix this. We don't document the LibGit2 module anywhere near as thoroughly as we need to, but yes it is exposed and some packages (e.g. Homebrew.jl) are going to need to interact with it.